File: sujeong/체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var lostt = lost.filter{ !reserve.contains($0) }
    var reservee = reserve.sorted().filter{ !lost.contains($0) }
    
    for r in reservee {
        if let i = lostt.index(of:r-1 < 1 ? 1 : r-1){
            lostt.remove(at:i)
        }else if let i = lostt.index(of:r+1 > n ? n : r+1){
            lostt.remove(at:i)
        }
    }
    
    return n - lostt.count
}

File: chaewon/week1/체육복.py
def solution(n, lost, reserve):
    _lost = list(set(lost).difference(reserve))
    _reserve = list(set(reserve).difference(lost))
    for i in _lost:
        if (i - 1 in _reserve):
            _reserve.remove(i-1)
        elif (i + 1 in _reserve):
            _reserve.remove(i+1)
        else:
            n -= 1
    return n